src/nvim/window.c

```diff
@@ -4037,7 +4037,11 @@ win_T *win_vert_neighbor(tabpage_T *tp, win_T *wp, bool up, long count)
   foundfr = wp->w_frame;
 
   if (wp->w_floating) {
-    return prevwin;
+    if (win_valid(prevwin) && prevwin != curwin && !prevwin->w_floating) {
```

Comment:
Shouln`t it be `prevwin != wp` here? Though actually the third check will imply the second, so can just remove `prevwin != curwin`